service/lxd.go

```diff
@@ -424,68 +382,55 @@ func defaultNetworkInterfacesFilter(clients map[string]lxd.InstanceServer, peer
 	return false, addresses
 }
 
-// GetUplinkInterfaces returns a map of peer name to slice of api.Network that may be used with OVN.
-func (s LXDService) GetUplinkInterfaces(ctx context.Context, bootstrap bool, peers []mdns.ServerInfo) (map[string][]api.Network, error) {
-	clients, networks, err := getClientsAndNetworks(s, ctx, bootstrap, peers)
-	if err != nil {
-		return nil, err
-	}
-
-	candidates := make(map[string][]api.Network)
-	for peer, nets := range networks {
-		for _, network := range nets {
-			filtered, addresses := defaultNetworkInterfacesFilter(clients, peer, network)
-			if filtered {
-				continue
-			}
-
-			if len(addresses) > 0 {
-				continue
-			}
-
-			candidates[peer] = append(candidates[peer], network)
-		}
-	}
-
-	return candidates, nil
-}
-
 // CephDedicatedInterface represents a dedicated interface for Ceph.
 type CephDedicatedInterface struct {
 	Name      string
 	Type      string
 	Addresses []string
 }
 
-// GetCephInterfaces returns a map of peer name to slice of CephDedicatedInterface that may be used to setup
-// a dedicated Ceph network.
-func (s LXDService) GetCephInterfaces(ctx context.Context, bootstrap bool, peers []mdns.ServerInfo) (map[string][]CephDedicatedInterface, error) {
-	clients, networks, err := getClientsAndNetworks(s, ctx, bootstrap, peers)
+// GetNetworkInterfaces fetches the list of networks from LXD and returns the following:
+// - A map of uplink compatible networks keyed by interface name.
+// - A map of ceph compatible networks keyed by interface name.
+// - The list of all networks.
+func (s LXDService) GetNetworkInterfaces(ctx context.Context, name string, address string, secret string) (map[string]api.Network, map[string]CephDedicatedInterface, []api.Network, error) {
+	var err error
+	var client lxd.InstanceServer
+	if name == s.Name() {
+		client, err = s.Client(ctx, "")
+	} else {
+		client, err = s.remoteClient(secret, address, CloudPort)
+	}
+
 	if err != nil {
-		return nil, err
+		return nil, nil, nil, err
 	}
 
-	candidates := make(map[string][]CephDedicatedInterface)
-	for peer, nets := range networks {
-		for _, network := range nets {
-			filtered, addresses := defaultNetworkInterfacesFilter(clients, peer, network)
-			if filtered {
-				continue
-			}
+	networks, err := client.GetNetworks()
+	if err != nil {
+		return nil, nil, nil, err
+	}
 
-			if len(addresses) == 0 {
-				continue
-			}
+	uplinkInterfaces := map[string]api.Network{}
+	cephInterfaces := map[string]CephDedicatedInterface{}
+	for _, network := range networks {
+		filtered, addresses := defaultNetworkInterfacesFilter(client, network)
+		if filtered {
+			continue
+		}
 
-			candidates[peer] = append(candidates[peer], CephDedicatedInterface{
+		if len(addresses) == 0 {
+			uplinkInterfaces[network.Name] = network
+		} else {
+			cephInterfaces[network.Name] = CephDedicatedInterface{
 				Name:      network.Name,
```

Comment:
Isn't the name redundant in here now?

Reply:
It may often be convenient to pass this around as a slice instead of a map, that's why the name is in both places, so we are not locked into always passing the map around.

Reply:
This way you don't have to give up the map structure which is already produced by `GetNetworkInterfaces()`.